### ISSUE TITLE
Support xlsx format for taxon files in DB Setup Wizard 

### ISSUE THREAD

**Issue by timo11** [specify/specify6#805]

**Comment by maxxxxxdlp** [comment]
Duplicate of #398

**Comment by timo11** [comment]
Wizard can now download and import taxon trees from xlsx files.

**Comment by maxxxxxdlp** [comment]
<img width="1440" alt="Screen Shot 2020-08-13 at 10 55 06 AM" src="https://user-images.githubusercontent.com/40512816/90108982-c37d1500-dd53-11ea-88c0-c758ec53f996.png">

Tree is empty
I did not see any related exceptions
I converted `.xls` file (that was working) to `.xlsx` via LibreOffice

Original file, new file, specify log, wizard log:
[Archive.zip](https://github.com/specify/specify6/files/5067836/Archive.zip)

**Comment by maxxxxxdlp** [comment]
Still does not work
[Archive.zip](https://github.com/specify/specify6/files/5070681/Archive.zip)

<img width="1440" alt="Screen Shot 2020-08-13 at 9 06 29 PM" src="https://user-images.githubusercontent.com/40512816/90170682-103f0b00-dda9-11ea-8795-e2d6ac9d5947.png">

@timo11 Can you send me any `.xlsx` files that worked for you?

**Comment by timo11** [comment]
[col2008_mammalia.xlsx](https://github.com/specify/specify6/files/5070695/col2008_mammalia.xlsx)
Here's one, in xlxs format, if it works.

**Comment by maxxxxxdlp** [comment]
@timo11 Nope, that did not work
<img width="1440" alt="Screen Shot 2020-08-13 at 9 17 21 PM" src="https://user-images.githubusercontent.com/40512816/90171585-61033380-ddaa-11ea-90f8-6abb9369c058.png">

XML file looks like this:
![Screen Shot 2020-08-13 at 9 17 43 PM](https://user-images.githubusercontent.com/40512816/90171629-6fe9e600-ddaa-11ea-801d-06e22f365e5f.png)

**Comment by timo11** [comment]
That file works for me. I just verified that I didn't disable downloading for testing or something.
Do you see the download dlg after leaving the taxon setup tab?
Do you see the Tree Building progress bar for very long during the build process?
If the file specified is already in the working directory -  '/home/[username]/Specify/' by default on linux - it won't be downloaded, if that's possibly any help.

**Comment by timo11** [comment]
Maybe it has something to do with using mammalogy (with Phylum) for a botany db (with Division)?

**Comment by maxxxxxdlp** [comment]
I think I found the problem

I get this screen and can select the file
<img width="965" alt="Screen Shot 2020-08-13 at 10 53 07 PM" src="https://user-images.githubusercontent.com/40512816/90180892-4b950600-ddb8-11ea-8f82-846a1453fd30.png">

Though, the file is not downloaded from the server
<img width="885" alt="Screen Shot 2020-08-13 at 10 56 33 PM" src="https://user-images.githubusercontent.com/40512816/90180915-53ed4100-ddb8-11ea-97e4-a501db2c921e.png">

It is not downloaded even if I change it's name

SpWizard does not report anything during this process
<img width="1108" alt="Screen Shot 2020-08-13 at 10 56 47 PM" src="https://user-images.githubusercontent.com/40512816/90181084-90b93800-ddb8-11ea-9cbc-2d6bfa356a6b.png">

I am deleting my `~/Documents/Specify` and `~/Specify` folders before opening the Wizard, but still, the file is not downloaded from the server

**Comment by timo11** [comment]
Do xls files get downloaded? They do for me from the default server using the new xml config file name.

**Comment by maxxxxxdlp** [comment]
> Maybe it has something to do with using mammalogy (with Phylum) for a botany db (with Division)?

Nope. That file is never downloaded. Not even for mammology
That one line in the log is me downloading the file via browser to see that the server works fine
<img width="1094" alt="Screen Shot 2020-08-13 at 11 02 35 PM" src="https://user-images.githubusercontent.com/40512816/90181427-176e1500-ddb9-11ea-82a9-2c099a1fb81d.png">

Also, I tested it a few days ago and came to the conclusion that changing Phylum for Division and vise versa does not have any negative impact

**Comment by maxxxxxdlp** [comment]
`.xls` gets downloaded properly. I can even see a briew pop up window while that file was downloading
<img width="848" alt="Screen Shot 2020-08-13 at 11 10 02 PM" src="https://user-images.githubusercontent.com/40512816/90182110-1f7a8480-ddba-11ea-830a-1fb7ec583d91.png">

<img width="947" alt="Screen Shot 2020-08-13 at 11 09 50 PM" src="https://user-images.githubusercontent.com/40512816/90182163-34571800-ddba-11ea-9b36-850531221884.png">

But `.xlsx` file does not get requested from the server

**Comment by maxxxxxdlp** [comment]
If it helps, this is the command I use to run the server: 
```
sudo python3 -m http.server 80
```

And this the relevant part from my `/etc/hosts`:
```
127.0.0.1	files.specifysoftware.org
```

**Comment by timo11** [comment]
It should work now. The downloader was only being called if the selected file had an "xls" extension.

**Comment by maxxxxxdlp** [comment]
`.xlsx` files are now downloaded from the server, but the tree is still empty

<img width="775" alt="Screen Shot 2020-08-14 at 11 02 11 AM" src="https://user-images.githubusercontent.com/40512816/90227808-d65c1c00-de1d-11ea-8b41-760f4406c868.png">

<img width="1440" alt="Screen Shot 2020-08-14 at 10 58 49 AM" src="https://user-images.githubusercontent.com/40512816/90227845-e2e07480-de1d-11ea-867e-4f82d4e2cf5a.png">

[Archive.zip](https://github.com/specify/specify6/files/5073324/Archive.zip)

**Comment by maxxxxxdlp** [comment]
@timo11 Sorry, I think I got confused yesterday. Wizard taxa import works with `.xls`, but does not work with `.xlsx` (it produces an empty tree with no errors in the console, even though the file is downloaded from the server)

**Comment by maxxxxxdlp** [comment]
@timo11, I tested this again
XLS works fine
The same file converted to XLSX does not work (even though it is downloaded from the server)
I also used the XLSX file that @timo11 used for testing, but it did not work for me either
[Archive.zip](https://github.com/specify/specify6/files/5285080/Archive.zip)


<img width="832" alt="Screen Shot 2020-09-25 at 10 58 58 PM" src="https://user-images.githubusercontent.com/40512816/94310874-ff2a1200-ff82-11ea-842a-38bce1faa99c.png">

**Comment by yvonneekanim** [comment]
@maxxxxxdlp @timo11 which folder should the new taxon file be in?

**Comment by maxxxxxdlp** [comment]
@yvonneekanim oh, testing this one is kind of complicated and would require setting up a web server. I would try to do that later today.
In the meantime, you can test the `.xls` and `.xlsx` support in other parts of Specify, like mentioned here - https://github.com/specify/specify6/issues/398#issuecomment-706400825

**Comment by maxxxxxdlp** [comment]
@timo11 I have just tried it on the latest build from the `develop` branch, and it didn't work for me.

I used the spreadsheet you used for testing.

`taxonfiles_2.xml`:

![Screenshot_2021-05-31_18-51-48](https://user-images.githubusercontent.com/40512816/120217791-854cd580-c241-11eb-95df-675c9e7c595c.png)

File is in the `taxonfiles/` directory on the server:

![Screenshot_2021-05-31_18-52-21](https://user-images.githubusercontent.com/40512816/120217797-867e0280-c241-11eb-99ca-946cae5ac8bf.png)

NGINX says that the file was downloaded succesfully
![Screenshot_2021-05-31_18-50-59](https://user-images.githubusercontent.com/40512816/120217836-91d12e00-c241-11eb-894e-1b2eab6df2f5.png)

And the downloaded file is present on my system:

![Screenshot_2021-05-31_18-56-24](https://user-images.githubusercontent.com/40512816/120218102-ebd1f380-c241-11eb-84f0-244c23c1c50a.png)

Yet the taxonomy tree is empty:
![Screenshot_2021-05-31_18-52-55](https://user-images.githubusercontent.com/40512816/120217800-87169900-c241-11eb-8509-8fb3171cd2f0.png)

**Comment by timo11** [comment]
The wiz should now download and load xlsx files. I don't think anybody but Max can verify this as of now. In any case, we should probably wait on testing this till we get new authority files.

**Comment by maxxxxxdlp** [comment]
Worked when I converted an old xls file to xlsx using libreoffice.
Did not work on this Animalia file generated by the tool:
[itis.zip](https://github.com/specify/specify6/files/6760331/itis.zip)

Yet, when I trimmed it down to just 20 rows, it worked fine:
[itis_trimmed.zip](https://github.com/specify/specify6/files/6760334/itis_trimmed.zip)

Taking the same 20 rows and duplicating them till I got a million rows caused the wizard to break again:
[itis_duplicated.zip](https://github.com/specify/specify6/files/6760354/itis_duplicated.zip)

Thus there seems to be some kind of a row limit.

**Comment by maxxxxxdlp** [comment]
Taxa file with 100k rows works fine.
Taxa file with 130k rows does not work.
No exception is thrown. Just this message is shown:
![Screenshot_2021-07-04_19-12-53](https://user-images.githubusercontent.com/40512816/124841336-a8337d80-df95-11eb-9b07-36917e06d5fa.png)

**Comment by maxxxxxdlp** [comment]
Here are the results after #968:
Taxa file with 230k rows works fine.
The entire Animalia (550k rows) does not work
Seems like this was a memory issue after all.
